### ISSUE TITLE
spicy-runtime package generates incorrect version

### ISSUE THREAD

**Issue by bbannier** [zeek/spicy#669]
Currently the `zeek-packages` CI task fails since it cannot compute the Spicy version,

```
/root/.zkg/clones/package/spicy-runtime/build/spicy/src/spicy/scripts/autogen-version: line 114: 3a4d367: value too great for base (error token is "3a4d367")
```

What is interesting is that `3a4d367` is not a SHA of known Spicy commit, but appears to refer to https://github.com/zeek/spicy-runtime/commit/3a4d367b0c72fdb8bb460a7c2220b13d3be0b197. This points to us executing Spicy's `autogen-version` in an incorrect directory (in a `spicy-runtime` instead of `spicy` checkout). The fix is probably to configure a correct directory in the CMake setup of the package.

**Comment by bbannier** [comment]
Configuring the external project with `BUILD_IN_SOURCE ON` allows `autogen-version` to pass, but we run into other issues during the build,

```
In file included from /root/.zkg/testing/spicy-runtime/clones/spicy-runtime/build/spicy/src/spicy/zeek/plugin/src/file-analyzer.cc:5:
/root/.zkg/testing/spicy-runtime/clones/spicy-runtime/build/spicy/src/spicy/zeek/plugin/include/zeek-spicy/plugin.h:14:10: fatal error: compiler/driver.h: No such file or directory
   14 | #include <compiler/driver.h>
      |          ^~~~~~~~~~~~~~~~~~~
```

**Comment by rsmmr** [comment]
Fixed by #675
